frontend/appflowy_tauri/src/appflowy_app/views/DocumentPage.hooks.ts

```diff
@@ -7,33 +7,40 @@ import {
 import { useParams } from 'react-router-dom';
 import { DocumentData } from '../interfaces/document';
 import { DocumentController } from '$app/stores/effects/document/document_controller';
-
+import { useAppDispatch } from '../stores/store';
 
 export const useDocument = () => {
   const params = useParams();
-  const [ documentId, setDocumentId ] = useState<string>();
-  const [ documentData, setDocumentData ] = useState<DocumentData>();
-  const [ controller, setController ] = useState<DocumentController | null>(null);
+  const [documentId, setDocumentId] = useState<string>();
+  const [documentData, setDocumentData] = useState<DocumentData>();
+  const [controller, setController] = useState<DocumentController | null>(null);
+  const dispatch = useAppDispatch();
 
   useEffect(() => {
+    let document_controller: DocumentController | null = null;
     void (async () => {
       if (!params?.id) return;
-      const c = new DocumentController(params.id);
-      setController(c);
+      console.log('==== enter ====', params?.id);
+      document_controller = new DocumentController(params.id, dispatch);
+      setController(document_controller);
       try {
-        const res = await c.open();
-        console.log(res)
+        const res = await document_controller.open();
+        console.log(res);
         if (!res) return;
         setDocumentData(res);
         setDocumentId(params.id);
       } catch (e) {
-        console.log(e)
+        console.log(e);
```

Comment:
Not sure if the console logs in this should be removed, but this one in the catch could be `console.error`